pkg/config/db.go

```diff
@@ -0,0 +1,156 @@
+	body.SharedBuffers = a.Settings.SharedBuffers
+
+	// Parameters that can be changed without restart
+	body.EffectiveCacheSize = a.Settings.EffectiveCacheSize
+	body.LogicalDecodingWorkMem = a.Settings.LogicalDecodingWorkMem
+	body.MaintenanceWorkMem = a.Settings.MaintenanceWorkMem
+	body.MaxLocksPerTransaction = cast.UintToIntPtr(a.Settings.MaxLocksPerTransaction)
+	body.MaxParallelMaintenanceWorkers = cast.UintToIntPtr(a.Settings.MaxParallelMaintenanceWorkers)
+	body.MaxParallelWorkersPerGather = cast.UintToIntPtr(a.Settings.MaxParallelWorkersPerGather)
+	body.MaxSlotWalKeepSize = a.Settings.MaxSlotWalKeepSize
+	body.MaxStandbyArchiveDelay = a.Settings.MaxStandbyArchiveDelay
+	body.MaxStandbyStreamingDelay = a.Settings.MaxStandbyStreamingDelay
+	body.MaxWalSize = a.Settings.MaxWalSize
+	body.SessionReplicationRole = (*v1API.UpdatePostgresConfigBodySessionReplicationRole)(a.Settings.SessionReplicationRole)
+	body.StatementTimeout = a.Settings.StatementTimeout
+	body.WalKeepSize = a.Settings.WalKeepSize
+	body.WalSenderTimeout = a.Settings.WalSenderTimeout
+	body.WorkMem = a.Settings.WorkMem
+	return body
+}
+
+func (a *db) fromRemoteDbConfig(remoteConfig v1API.PostgresConfigResponse) db {
+	result := *a
+
+	result.Settings.EffectiveCacheSize = remoteConfig.EffectiveCacheSize
+	result.Settings.LogicalDecodingWorkMem = remoteConfig.LogicalDecodingWorkMem
+	result.Settings.MaintenanceWorkMem = remoteConfig.MaintenanceWorkMem
+	result.Settings.MaxConnections = cast.IntToUintPtr(remoteConfig.MaxConnections)
+	result.Settings.MaxLocksPerTransaction = cast.IntToUintPtr(remoteConfig.MaxLocksPerTransaction)
+	result.Settings.MaxParallelMaintenanceWorkers = cast.IntToUintPtr(remoteConfig.MaxParallelMaintenanceWorkers)
+	result.Settings.MaxParallelWorkers = cast.IntToUintPtr(remoteConfig.MaxParallelWorkers)
+	result.Settings.MaxParallelWorkersPerGather = cast.IntToUintPtr(remoteConfig.MaxParallelWorkersPerGather)
+	result.Settings.MaxReplicationSlots = cast.IntToUintPtr(remoteConfig.MaxReplicationSlots)
+	result.Settings.MaxSlotWalKeepSize = remoteConfig.MaxSlotWalKeepSize
+	result.Settings.MaxStandbyArchiveDelay = remoteConfig.MaxStandbyArchiveDelay
+	result.Settings.MaxStandbyStreamingDelay = remoteConfig.MaxStandbyStreamingDelay
+	result.Settings.MaxWalSenders = cast.IntToUintPtr(remoteConfig.MaxWalSenders)
+	result.Settings.MaxWalSize = remoteConfig.MaxWalSize
+	result.Settings.MaxWorkerProcesses = cast.IntToUintPtr(remoteConfig.MaxWorkerProcesses)
+	result.Settings.SessionReplicationRole = (*string)(remoteConfig.SessionReplicationRole)
+	result.Settings.SharedBuffers = remoteConfig.SharedBuffers
+	result.Settings.StatementTimeout = remoteConfig.StatementTimeout
+	result.Settings.WalKeepSize = remoteConfig.WalKeepSize
+	result.Settings.WalSenderTimeout = remoteConfig.WalSenderTimeout
+	result.Settings.WorkMem = remoteConfig.WorkMem
+	return result
+}
+
+func (a *db) DiffWithRemote(remoteConfig v1API.PostgresConfigResponse) ([]byte, error) {
```

Comment:
Perhaps one more suggestion is to make these methods of `settings` struct so that it's cleaner to support syncing `pooler` config in the future.

```suggestion
func (a *settings) DiffWithRemote(remoteConfig v1API.PostgresConfigResponse) ([]byte, error) {
```